airflow/providers/ssh/operators/ssh.py

```diff
@@ -18,15 +18,16 @@
 
 import warnings
 from base64 import b64encode
-from select import select
-from typing import Optional, Sequence, Tuple, Union
-
-from paramiko.client import SSHClient
+from typing import TYPE_CHECKING, Optional, Sequence, Union
 
 from airflow.configuration import conf
 from airflow.exceptions import AirflowException
 from airflow.models import BaseOperator
-from airflow.providers.ssh.hooks.ssh import SSHHook
+
+if TYPE_CHECKING:
+    from paramiko.client import SSHClient
+
+    from airflow.providers.ssh.hooks.ssh import SSHHook
```

Comment:
For my own edification, why only import this when type checking?

Reply:
Yeah. My question too.

Reply:
Not strictly required, it just means DAG parsing has to import less.

Reply:
Hmm. Sounds like new patern in the making. Some entirely discussion to make, but this touches some interesting possibilities.